packages/slice-machine/lib/builders/SliceBuilder/index.tsx

```diff
@@ -74,38 +106,42 @@ const SliceBuilder: React.FunctionComponent = () => {
     }
   }, [data]);
 
-  useEffect(() => {
-    if (!store) return;
-
-    return () => store.reset();
-  }, []);
-
   const sliceView = useMemo(
-    () => [{ sliceID: Model.model.id, variationID: variation.id }],
-    [Model.model.id, variation.id]
+    () =>
+      Model && variation
+        ? [{ sliceID: Model.model.id, variationID: variation.id }]
+        : null,
+    [Model?.model.id, variation?.id]
   );
 
+  if (!Model || !variation || !sliceView) return null;
+
   const onTakingCustomScreenshot = () => {
-    checkSimulatorSetup(true, () =>
-      // eslint-disable-next-line @typescript-eslint/no-unsafe-return
-      store
-        .variation(variation.id)
-        .generateScreenShot(Model.from, Model.model.name, setData)
-    );
+    // eslint-disable-next-line @typescript-eslint/no-misused-promises
+    checkSimulatorSetup(true, async () => {
+      await generateScreenShot(
+        variation.id,
+        Model.from,
+        Model.model.name,
+        setData,
+        (screenshots) => generateSliceScreenshot(screenshots)
+      );
+    });
   };
 
   return (
     <Box sx={{ flex: 1 }}>
       <Header
         Model={Model}
-        store={store}
         variation={variation}
-        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-        // @ts-expect-error
-        onPush={() => void store.push(Model, onPush)}
-        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-        // @ts-expect-error
-        onSave={() => void store.save(Model, setData)}
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment, @typescript-eslint/no-misused-promises
```

Comment:
we return a Promise<void> rather than a void. Annoying to change however since we'd have to type everything as a Promise<void> instead, and as some point this type is passed to the onClick prop of a theme-ui component, whose signature we can't change